cobalt/build/build_info.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python
+# Copyright 2023 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generates a Cobalt Build Info json."""
+
+import datetime
+import json
+import os
+import re
+import subprocess
+import sys
+
+_FILE_DIR = os.path.dirname(__file__)
+_COMMIT_COUNT_BUILD_ID_OFFSET = 1000000
+
+_BUILD_ID_PATTERN = '^BUILD_NUMBER=([1-9][0-9]{6,})$'
```

Comment:
I believe the `{6,}` pattern will not work for `git log --grep` which uses a subset of full regex rules by default.

Reply:
Oscar and I went back and forth on this, but this has been tested to work because with the `-E` flag `git log --grep` supports the extended regex rules.

Reply:
Ah, didn't see we were using extended regexp, that sg. Though still consider my other comments of reusing logic from get_build_id.py